File: backend/src/WasteToValue.Api/Modules/Collections/Entities/PickupEvent.cs
namespace WasteToValue.Api.Modules.Collections.Entities;

public sealed class PickupEvent
{
    public Guid Id { get; set; }
    public Guid PickupRequestId { get; set; }
    public string EventType { get; set; } = string.Empty;
    public Guid ActorId { get; set; }
    public DateTimeOffset EventAt { get; set; }
    public string? Notes { get; set; }
    public string IdempotencyKey { get; set; } = string.Empty;

    public PickupRequest? PickupRequest { get; set; }
}

File: mobile/lib/features/collections/models/pickup_event.dart
class PickupEvent {
  const PickupEvent({
    required this.id,
    required this.pickupRequestId,
    required this.eventType,
    required this.actorId,
    required this.eventAt,
    this.notes,
  });

  final String id;
  final String pickupRequestId;
  final String eventType;
  final String actorId;
  final DateTime eventAt;
  final String? notes;

  factory PickupEvent.fromJson(Map<String, dynamic> json) {
    return PickupEvent(
      id: json['id'] as String,
      pickupRequestId: json['pickupRequestId'] as String,
      eventType: json['eventType'] as String,
      actorId: json['actorId'] as String,
      eventAt: DateTime.parse(json['eventAt'] as String),
      notes: json['notes'] as String?,
    );
  }
}
